src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -94,13 +99,13 @@ impl QueryManager {
             .get_epoch(query_id.clone())
             .await?;
 
-        let creat_task_resp = compute_client
+        compute_client
             .create_task(task_id.clone(), plan, epoch)
-            .await;
+            .await
+            .map_err(|e| anyhow!(e))?;
         self.hummock_snapshot_manager
             .unpin_snapshot(epoch, &query_id)
             .await?;
-        creat_task_resp?;
```

Comment:
We should keep the original flow. Otherwise if create task failed, the snapshot will not be unpinned.